src/libcharon/sa/ikev2/tasks/ike_reauth.c

```diff
@@ -62,6 +62,9 @@ METHOD(task_t, process_i, status_t,
 	}
 
 	/* we always destroy the obsolete IKE_SA */
+#ifdef USE_KERNEL_SUPER
+	charon->kernel->write_terminator_tunnel_ike_sa_auth_uptime(charon->kernel, this->ike_sa);
```

Comment:
1. `ike_reauth` is called only for the initiator of the reauthentication. The responder will not have this called.
2. `ike_reauth` is used only for break-before-make reauthentication. In case of make-before-break a different flow is used (see `task_manager_v2.c :: queue_ike_reauth()`)